pkg/kubelet/cm/dra/plugin/registration.go

```diff
@@ -287,52 +300,63 @@ func (h *RegistrationHandler) DeRegisterPlugin(pluginName, endpoint string) {
-		// in the context of the registration handler, the one from
-		// the plugin is canceled.
-		logger = klog.FromContext(h.backgroundCtx)
-		logger = klog.LoggerWithName(logger, "driver-cleanup")
-		logger = klog.LoggerWithValues(logger, "pluginName", pluginName)
-		ctx, cancel := context.WithCancelCause(h.backgroundCtx)
-		ctx = klog.NewContext(ctx, logger)
-
-		// Clean up the ResourceSlices for the deleted Plugin since it
-		// may have died without doing so itself and might never come
-		// back.
-		//
-		// May get canceled if the plugin comes back quickly enough
-		// (see RegisterPlugin).
-		h.mutex.Lock()
-		defer h.mutex.Unlock()
-		if cancel := h.pendingWipes[pluginName]; cancel != nil {
-			(*cancel)(errors.New("plugin deregistered a second time"))
-		}
-		h.pendingWipes[pluginName] = &cancel
-
-		h.wg.Add(1)
-		go func() {
-			defer h.wg.Done()
-			defer func() {
-				h.mutex.Lock()
-				defer h.mutex.Unlock()
-
-				// Cancel our own context, but remove it from the map only if it
-				// is the current entry. Perhaps it already got replaced.
-				cancel(errors.New("wiping done"))
-				if h.pendingWipes[pluginName] == &cancel {
-					delete(h.pendingWipes, pluginName)
-				}
-			}()
-			h.wipeResourceSlices(ctx, h.wipingDelay, pluginName)
-		}()
-		return
+		h.cleanupResourceSlices(pluginName)
 	}
 
 	logger := klog.FromContext(h.backgroundCtx)
 	logger.V(3).Info("Deregister DRA plugin not necessary, was already removed")
 }
 
+// cleanupResourceSlices initiates the cleanup process for ResourceSlices
+// associated with a given plugin. It prepares a new context and logger for
+// the cleanup operation, ensuring that any previous cleanup for the same
+// plugin is canceled if still pending. The cleanup is performed
+// asynchronously in a background goroutine, which will remove the
+// ResourceSlices after a delay unless the plugin is back online before the
+// cleanup completes.
+func (h *RegistrationHandler) cleanupResourceSlices(pluginName string) {
+	// Prepare for canceling the background wiping. This needs to run
+	// in the context of the registration handler, the one from
+	// the plugin is canceled.
+	logger := klog.FromContext(h.backgroundCtx)
+	logger = klog.LoggerWithName(logger, "driver-cleanup")
+	logger = klog.LoggerWithValues(logger, "pluginName", pluginName)
+
+	logger.V(3).Info("Starting ResourceSlices cleanup", "pluginName", pluginName)
+	ctx, cancel := context.WithCancelCause(h.backgroundCtx)
+	ctx = klog.NewContext(ctx, logger)
+
+	// Clean up the ResourceSlices for the Plugin since it
+	// may have died without doing so itself and might never come
+	// back.
+	//
+	// May get canceled if the plugin comes back quickly enough
+	// (see RegisterPlugin).
+	h.mutex.Lock()
+	defer h.mutex.Unlock()
+	if cancel := h.pendingWipes[pluginName]; cancel != nil {
+		(*cancel)(errors.New("plugin deregistered a second time"))
+	}
+	h.pendingWipes[pluginName] = &cancel
```

Comment:
@pohly I suspect this code shouldn't cancel ongoing wiping, but let it continue. How about doing something like this at the beginning of this function?
```
if cancel := h.pendingWipes[pluginName]; cancel != nil {
        logger.V(3).Info("Previously started cleanup detected, do nothing", "pluginName", pluginName)
	return
}
```